source/packages/libraries/core/version/scripts/compile.ts

```diff
@@ -0,0 +1,48 @@
+import fs from 'fs';
+import readline from 'readline';
+import { Project } from 'ts-morph';
+
+async function extractVersion(): Promise<string> {
+    const versionPoliciesReadStream = fs.createReadStream(
+        '../../../../common/config/rush/version-policies.json'
+    );
+    const rl = readline.createInterface({
+        input: versionPoliciesReadStream,
+        crlfDelay: Infinity,
+    });
+
+    let previousLine: string = '';
+    let versionLine: string | undefined = undefined;
+    for await (const line of rl) {
+        if (previousLine.includes('"policyName": "external",')) {
+            versionLine = line;
+            break;
+        }
+
+        previousLine = line;
+    }
+
+    // https://semver.org/#is-there-a-suggested-regular-expression-regex-to-check-a-semver-string
+    const versionRegex =
+        /(0|[1-9]\d*)\.(0|[1-9]\d*)\.(0|[1-9]\d*)(?:-((?:0|[1-9]\d*|\d*[a-zA-Z-][0-9a-zA-Z-]*)(?:\.(?:0|[1-9]\d*|\d*[a-zA-Z-][0-9a-zA-Z-]*))*))?(?:\+([0-9a-zA-Z-]+(?:\.[0-9a-zA-Z-]+)*))?/;
+    const version = versionLine!.match(versionRegex);
```

Comment:
This can error if no version line is found. Should add an error that indicates what happened.

Reply:
Same question as below - it's not clear to me when can this possibly happen. This entire util package is built on top of the assumption that "there will be a `version-policies.json` managed/generated by rush that will _always_ exists with the version we need" (I do think I should add this in the comment).